pkg/cap/cluster/anchor/host.go

```diff
@@ -0,0 +1,194 @@
+package anchor
+
+import (
+	"context"
+	"sync"
+
+	"capnproto.org/go/capnp/v3"
+	"capnproto.org/go/capnp/v3/rpc"
+	"capnproto.org/go/capnp/v3/server"
+	"github.com/libp2p/go-libp2p-core/peer"
+	api "github.com/wetware/ww/internal/api/cluster"
+	"github.com/wetware/ww/pkg/cap/cluster"
+	"github.com/wetware/ww/pkg/vat"
+)
+
+var (
+	Capability = vat.BasicCap{
+		"hostAnchor/packed",
+		"hostAnchor"}
+	defaultPolicy = server.Policy{
+		// HACK:  raise MaxConcurrentCalls to mitigate known deadlock condition.
+		//        https://github.com/capnproto/go-capnproto2/issues/189
+		MaxConcurrentCalls: 64,
+		AnswerQueueSize:    64,
+	}
+)
+
+type HostAnchor struct {
+	Peer peer.ID
+	Vat  vat.Network
+
+	Client api.Host
+
+	once sync.Once
+}
+
+func (ha HostAnchor) Path() []string {
+	return []string{ha.Peer.String()}
+}
+
+func (ha HostAnchor) Ls(ctx context.Context) (AnchorIterator, error) {
+	if err := ha.bootstrapOnce(ctx); err != nil {
+		return nil, err
+	}
+
+	anchor := api.Anchor{Client: ha.Client.Client}
+	return newIterator(ctx, anchor, ha.Path())
+}
+
+func (ha HostAnchor) Walk(ctx context.Context, path []string) (Anchor, error) {
+	if err := ha.bootstrapOnce(ctx); err != nil {
+		return nil, err
+	}
+
+	fut, release := ha.Client.Walk(ctx, func(a api.Anchor_walk_Params) error {
+		capPath, err := a.NewPath(int32(len(path)))
+		if err != nil {
+			return err
+		}
+		for i, e := range path {
+			if err := capPath.Set(i, e); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+
+	return ContainerAnchor{path: append(ha.Path(), path...), fut: fut, release: release}, nil
+}
+
+func (ha HostAnchor) bootstrapOnce(ctx context.Context) error {
+	var (
+		conn *rpc.Conn
+		err  error
+	)
+
+	if ha.Client.Client != nil {
+		return nil
+	}
+
+	ha.once.Do(func() {
+		conn, err = ha.Vat.Connect(
+			ctx,
+			peer.AddrInfo{ID: ha.Peer},
+			Capability,
+		)
+		if err != nil {
+			return
+		}
+		ha.Client = api.Host{Client: conn.Bootstrap(ctx)}
+	})
+
+	return err
+}
+
+type HostAnchorIterator struct {
+	Vat     vat.Network
+	It      *cluster.Iterator
+	Release capnp.ReleaseFunc
+}
+
+func (hai HostAnchorIterator) Next(ctx context.Context) error {
+	hai.It.Next(ctx)
+	return hai.It.Err
+}
+
+func (hai HostAnchorIterator) Finish() {
+	hai.Release()
+}
+
+func (hai HostAnchorIterator) Anchor() Anchor {
+	return HostAnchor{
+		Peer: hai.It.Record().Peer(),
+		Vat:  hai.Vat,
+	}
+}
+
+type HostAnchorServer struct {
+	vat vat.Network
+
+	tree *node
+
+	client api.Host
```

Comment:
The name `client` is misleading here.  If anything, it's a server, but perhaps you just want to call this `host` or `h`?

Reply:
The thing is that that's the Anchor capability client that will be passed around the network. So, why shouldn't we call it client? Not sure why you consider it a server.

Reply:
Oh, you're right.  My mistake!  (I got this confused with `api.Host_Server`)